timemachine/fe/rbfe.py

```diff
@@ -125,34 +85,70 @@ def __init__(self, omm_system, conf, box):
         self.box = box
 
 
-def generate_samples(st, host_config, temperature, lambda_schedule, n_frames, seed):
+def sample(initial_state, protocol):
+    """
+    Generate a trajectory given an initial state and a simulation protocol
     """
-    Generate samples given a lambda_schedule. Let L be the number
-    of lambda windows T be the number of frames.
 
-    Parameters
-    ----------
-    st: SingleTopology
-        Topology container for the alchemical ligand
+    bound_impls = [p.bound_impl(np.float32) for p in initial_state.U_fns]
+    intg_impl = initial_state.integrator.impl()
+    baro_impl = initial_state.barostat.impl(bound_impls)
 
-    host_config: HostConfig
-        Configuration of the host
+    ctxt = custom_ops.Context(initial_state.x0, initial_state.v0, initial_state.box0, intg_impl, bound_impls, baro_impl)
 
-    temperature: float
-        Temperature of the system, in Kelvins
+    # burn-in
+    ctxt.multiple_steps_U(
+        lamb=initial_state.lamb,
+        n_steps=protocol.burn_in,
+        lambda_windows=[initial_state.lamb],
+        store_u_interval=0,
+        store_x_interval=0,
+    )
 
-    lambda_schedule: np.array(L,)
-        Sequence of lambda windows
+    # a crude, and probably not great, guess on the decorrelation time
+    n_steps = protocol.n_frames * protocol.steps_per_frame
+    all_nrgs, all_coords, all_boxes = ctxt.multiple_steps_U(
+        lamb=initial_state.lamb,
+        n_steps=n_steps,
+        lambda_windows=[initial_state.lamb],
+        store_u_interval=protocol.steps_per_frame,
+        store_x_interval=protocol.steps_per_frame,
+    )
 
-    n_frames: int
-        Number of samples to generate per window
+    assert all_coords.shape[0] == protocol.n_frames
+    assert all_boxes.shape[0] == protocol.n_frames
+
+    return all_coords, all_boxes
+
+
+class SimulationProtocol:
+    def __init__(self, n_frames, burn_in=10000, steps_per_frame=1000):
+        self.n_frames = n_frames
+        self.burn_in = burn_in
+        self.steps_per_frame = steps_per_frame
 
-    Returns
-    -------
-    3-tuple of all_frames, all_boxes, all_U_fns
-        all_frames has shape (L,T,N,3), all_boxes has shape (L,T,3,3), all_U_fns has shape (L,)
 
+class InitialState:
```

Comment:
"State" seems like a pretty overloaded term. I wonder if there's something more specific we could use here? One thought is something like `TrajectorySeed` (I initially equated this with "thermodynamic state / potential" and was confused about where the rng seed was being stored)

Reply:
yeah I agree - I'll rename this